scripts/morty/morty_pde_ode_compare.py

```diff
@@ -0,0 +1,413 @@
+        nu_vec : 1D vector
+            Flow rate terms over space
+        mu_vec : 1D vector
+            Loss terms over space
+
+        """
+        S_vec = self._format_spatial(self.S1, self.S2, False)
+        nu_vec = self._format_spatial(self.nu1, self.nu2, False)
+        mu_vec = self._format_spatial(self.mu1, self.mu2, False)
+        return S_vec, nu_vec, mu_vec
+
+    def _vary_source(self):
+        """
+        The source and loss terms are treated as a sinusoidal
+        profile in the in-core region rather than as a constant.
+
+        Returns
+        -------
+        S_vec : 1D vector
+            Source terms over space
+        nu_vec : 1D vector
+            Flow rate terms over space
+        mu_vec : 1D vector
+            Loss terms over space
+
+        """
+        core_mod = np.pi/2 * np.sin(self.zs * np.pi / self.z1)
+        S_vec = self._format_spatial(self.S1*core_mod, self.S2,
+                                     vector_form=True)
+        nu_vec = self._format_spatial(self.nu1, self.nu2)
+        mu_vec = self._format_spatial(self.mu1*core_mod, self.mu2,
+                                      vector_form=True)
+        return S_vec, nu_vec, mu_vec
+
+    def fd_PDE(self, spatial_vary):
+        """
+        Solves the PDE using finite difference explicit in time and
+        upwind (backwards) in space.
+
+        Parameters
+        ----------
+        spatial_vary : bool
+            True means the source and loss terms will have a sinusoidal shape
+            in the in-core region.
+
+        Returns
+        -------
+        zs : 1D vector
+            Spatial nodes
+        ts : 1D vector
+            Time nodes
+        N_z_t : 2D matrix
+            Each row represents the spatial concentration
+            at a given point in time
+
+        """
+        conc = self.init_conc
+        if not spatial_vary:
+            S_vec, nu_vec, mu_vec = self._step_source()
+        else:
+            S_vec, nu_vec, mu_vec = self._vary_source()
+        J = np.arange(0, self.nodes)
+        Jm1 = np.roll(J,  1)
+
+        N_z_t = np.zeros((len(self.ts)+1, self.nodes))
+        N_z_t[0, :] = conc
+        conc_mult = 1 - mu_vec * self.dt
+        add_source = S_vec * self.dt
+        for ti, t in enumerate(self.ts):
+            conc = (add_source + conc_mult * conc
+                    + self.lmbda * (conc[Jm1] - conc))
+            N_z_t[ti+1, :] = conc
+
+        return self.zs, self.ts, N_z_t
+
+    def ODE(self):
+        """
+        Solves the ODE form of depletion (no neutronic updates)
+
+        Returns
+        -------
+        conc : list of 1D vectors
+            Each item in the list is the spatial concentration vector
+
+        """
+        conc = []
+        for t in ts:
+            current = (self.init_conc[0] * np.exp(-self.mu1 * t) +
+                       self.S1/self.mu1 * (1 - np.exp(-self.mu1 * t)))
+            conc.append(current)
+        return conc
+
+
+if __name__ == '__main__':
```

Comment:
This should be a separate "plotting" file. Especially since you use the `FormatAssist` class in another file.